File: tests/fixtures/edgecase_repo/csharp/Service3.cs
using System;

namespace Example.Gen
{
    public interface IHandler3
    {
        bool Handle(string payload);
    }

    public class Base3
    {
        protected string Name;
    }

    public class Service3 : Base3, IHandler3
    {
        public Service3(string name)
        {
            this.Name = name;
        }

        public bool Handle(string payload)
        {
            this.Validate(payload);
            return true;
        }

        private void Validate(string payload) { }
    }
}

File: tests/fixtures/edgecase_repo/kotlin/Service3.kt
package com.example.gen

import com.example.gen.Base3

interface Handler3 {
    fun handle(payload: String): Boolean
}

open class Base3(val name: String)

class Service3(name: String) : Base3(name), Handler3 {
    override fun handle(payload: String): Boolean {
        validate(payload)
        return true
    }

    private fun validate(payload: String) {}
}

fun helper3(value: String): String {
    return value
}
